ckan/views/__init__.py

```diff
@@ -43,25 +45,97 @@ def set_cors_headers_for_response(response: Response) -> Response:
     return response
 
 
+def set_etag_and_fast_304_response_if_unchanged(response: Response) -> Response:
+    """Set ETag and return 304 if content is unchanged."""
+
+    enable_etags = config.get(u'ckan.cache_etags', True)
+    enable_etags_not_modified = config.get(u'ckan.cache_etags_notModified', True)
+    etag = None
+
+    allowed_status_codes = {200, 404}
```

Comment:
Are constants available for these codes?

Also, are there any other codes in the 2xx range that we should recognise?

And why would there be ETags on 404?